contracts/fee_currencies.go

```diff
@@ -100,6 +100,8 @@ func DebitFees(evm *vm.EVM, feeCurrency *common.Address, address common.Address,
 				err,
 			)
 		}
+		// This might still be an out-of-gas error, since those are not propagated from sub-calls.
```

Comment:
```suggestion
		// This error could still be caused by lack of gas, since those are not always propagated from sub-calls.
```

I think this is a little clearer, the previous comment was a bit confusing because at this point the error definitely is not `vm.ErrOutOfGas`, same for below